src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,11 +23,127 @@ Copyright (c) .NET Foundation. All rights reserved.
+                 Properties="ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(_TFM);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             DisableImplicitFrameworkReferences=true;
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"  />
+   </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMapper">
+    
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="CacheWorker"
+                 Properties="CachePackageName=%(PackageReference.Identity);
+                             CachePackageVersion=%(PackageReference.Version);
+                             ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(TargetFramework);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"
+             />
+  </Target>
+
+
+  <Target Name="CacheWorker"
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepforRestoreForComposeCache;
+                            CacheWorkerPerformWork"/>
+
+
+  <Target Name="CacheWorkerPerformWork"
+          DependsOnTargets="RestoreForComposeCache;
+                            ComputeAndCopyFilesToCacheDirectory;"
+          Condition="!Exists($(CacheWorkerWorkingDir))" />
+    <!--
+    ============================================================
+                                        CacheFinalizer
+
+   Cleansup and produces artifacts after Completion of cache
+    ============================================================
+    -->
   
+  <UsingTask TaskName="GetListofResolvedPackages" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="CacheFinalizer">
+    
+    <GetListofResolvedPackages>
+    <Output TaskParameter="PackagesResolved" ItemName="ListOfPackages" />
+    </GetListofResolvedPackages>
 
+    <ItemGroup>
+      <ListOfPackageReference Include="@(ListOfPackages -> '%20%20%20%20%20&lt;PackageReference Include=&quot;%(Identity)&quot;  Version =&quot;%(Version)&quot;/&gt;')"/>
+    </ItemGroup>
+    <PropertyGroup>
+     <_CacheArtifactContent>
+      <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="$(MSBuildToolsVersion)">
```

Comment:
Why are we using an MSBuild project for this .xml file?  There is no reason we should be using an MSBuild project/format.  It is just a list of Package Ids and Versions, and we are just parsing it with an Xml parser.  We should avoid using MSBuild just to hold a list of data.